### ISSUE TITLE
Huge Memory Leaks on 8.5.x drivers

### ISSUE THREAD

**Issue by mubeda** [FirebirdSQL/NETProvider#997]
We build our blazor project using NHibernate to support several DBMS.  Some of our Unit test projects uses Firebird Embedded to test database interactions.  We never experience problems until we updated the nuget package from 8.0.1 to 8.5.X
TestHost  process memory consumption jumped from 2 GB to 36GB! By rolling back to 8.0.1 the memory problem is gone.

NHibernate 5.3.9
Nnet 5 project under Windows 10.

**Comment by cincuranet** [comment]
I need more than a generic description. Some piece of code that shows the behavior.

**Comment by mubeda** [comment]
Unfourtunaly i cannot share the code due NDA, but i will try to do a memory profile on the project and try to use this repository instead of the nuget package to get more detail info about the spot. I will ping again after the results.

**Comment by mubeda** [comment]
I had included memory profile screenshots of the same unit test running with 8.0.1 and 8.5. 3 latest driver. 
For firebird embedded connection we use fbclient.dll 3.0.3.32900.  I tried upgrading to fbclient.dll 3.0.7.33374, but same results.
Let me know if this helps some way,  if not, preparing a working example will take me some time.

[profiling.zip](https://github.com/FirebirdSQL/NETProvider/files/7187154/profiling.zip)
